### ISSUE TITLE
Windows ASIO issues

### ISSUE THREAD

**Issue by aelray** [pythonnet/pythonnet#1463]
### Environment

-   Pythonnet version: 2.5.2
-   Python version: 3.8.8
-   Operating System: Windows 10
-   .NET Runtime: 4.8

### Details

-   Describe what you were trying to get done.

    Accessing Sound card via ASIO driver using [python-sounddevice](https://pypi.org/project/sounddevice)

-   What commands did you run to trigger this issue? If you can provide a

```python
   import clr
   import sounddevice as sd

   print(sd.query_devices())
```

-   If there was a crash, please include the traceback here.

```powershell
  0 Microsoft Sound Mapper - Output, MME (0 in, 2 out)
< 1 Remote Audio, MME (0 in, 2 out)
  2 Primary Sound Driver, Windows DirectSound (0 in, 2 out)
  3 Remote Audio, Windows DirectSound (0 in, 2 out)
  4 Remote Audio, Windows WASAPI (0 in, 2 out)
  5 Speakers (HD Audio Headphone/Speakers), Windows WDM-KS (0 in, 2 out)
  6 Analogue 1 + 2 (fniusbwc4400), Windows WDM-KS (8 in, 0 out)
  7 Speakers (fniusbwr4400), Windows WDM-KS (0 in, 8 out)
```

without importing `clr` 

```python
   import sounddevice as sd

   print(sd.query_devices())
```

we get

```powershell
  0 Microsoft Sound Mapper - Output, MME (0 in, 2 out)
< 1 Remote Audio, MME (0 in, 2 out)
  2 Primary Sound Driver, Windows DirectSound (0 in, 2 out)
  3 Remote Audio, Windows DirectSound (0 in, 2 out)
  4 ASIO4ALL v2, ASIO (8 in, 8 out)
  5 Focusrite USB ASIO, ASIO (20 in, 20 out)
  6 Remote Audio, Windows WASAPI (0 in, 2 out)
  7 Speakers (HD Audio Headphone/Speakers), Windows WDM-KS (0 in, 2 out)
  8 Analogue 1 + 2 (fniusbwc4400), Windows WDM-KS (8 in, 0 out)
  9 Speakers (fniusbwr4400), Windows WDM-KS (0 in, 8 out)
```

so devices 4 and 5 are missing, i.e. the ASIO API is not available.  [python-sounddevice](https://pypi.org/project/sounddevice) relies on PortAudio and for some reason it cannot query any ASIO devices as soon as pythonnet is imported.

Funny workaround:
```python
   import sounddevice as sd
   import clr

   print(sd.query_devices())
```
gives:

```powershell
  0 Microsoft Sound Mapper - Output, MME (0 in, 2 out)
< 1 Remote Audio, MME (0 in, 2 out)
  2 Primary Sound Driver, Windows DirectSound (0 in, 2 out)
  3 Remote Audio, Windows DirectSound (0 in, 2 out)
  4 ASIO4ALL v2, ASIO (8 in, 8 out)
  5 Focusrite USB ASIO, ASIO (20 in, 20 out)
  6 Remote Audio, Windows WASAPI (0 in, 2 out)
  7 Speakers (HD Audio Headphone/Speakers), Windows WDM-KS (0 in, 2 out)
  8 Analogue 1 + 2 (fniusbwc4400), Windows WDM-KS (8 in, 0 out)
  9 Speakers (fniusbwr4400), Windows WDM-KS (0 in, 8 out)
```
That's probably due to the fact that PortAudio is initialized before `clr` where it can still query the ASIO API for devices.

**Comment by filmor** [comment]
This is very odd :)

Thank you first for the detailed report, I'll try to reproduce this locally.

Does this happen with the current master as well?

**Comment by aelray** [comment]
Issue seems to be resolved with pythonnet 3.0.1
